### DISCUSSION TITLE
A "very internal" scope

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#5258] [Q&A]
Is there a way to make an internal class not visible to an [InternalsVisibleTo] assembly and only ever accessible to the current assembly?

**Comment by DavidArno** [comment]
Don't use `[InternalsVisibleTo]`? Aside from that, no there isn't. You either expose all internals, or none.

**Comment by CyrusNajmabadi** [comment]
Yes. Use can analyzer to block it.

**Comment by teo-tsirpanis** [comment]
There is the `compilercontrolled` visibility, but only for members, not types.

**Reply by KarolKalbarczyk** [reply]
Is this something available in C#? Or only IL?

**Reply by teo-tsirpanis** [reply]
Only IL.
